### DISCUSSION TITLE
Creating a MongoDB/Mongoose user provider for auth using the access tokens (or other) guard (v6)

### DISCUSSION THREAD

**Discussion by benhepburnn** [adonisjs/core#4381] [Help] [answered]
Hello, I'm looking for help on how to create a custom user provider for Mongoose models. Ideally I would like to just make a custom user provider and still use the tokens guard, however if I need to make a custom guard that's fine.

I've read the docs on [custom auth guards](https://docs.adonisjs.com/guides/auth-custom-guards) and looked at the [lucid user provider source](https://github.com/adonisjs/auth/blob/develop/modules/access_tokens_guard/user_providers/lucid.ts) to try adapt it but it's too difficult to follow.

Thanks for your help.

**Comment by thetutlage** [comment, marked as answer]
Hello, I will be writing a guide for the same by the end of this week. So stay tuned!

**Reply by mohamadlounnas** [reply]
hi @thetutlage , any news?

**Reply by akshay-rwaltz** [reply]
Hi, Whats the latest on this one?

**Comment by mdsadiqueinam** [comment]
Do not restrict the session guard to use lucdModel, this is difficult when need to use mongodb, I am creating a project which needs mongodb but I am unable to use the available auth_guard due to the resriction of session guard to lucidmodel

**Reply by thetutlage** [reply]
You need to create a user provider. Check this blog post on how to do it.

https://adonisjs.com/blog/kysely-with-auth-session-guard
